docs/src/main/tut/typeclasses/arrow.md

```diff
@@ -0,0 +1,119 @@
+- Composing `fab: F[A, B]` and `fbc: F[B, C]` into `fac: F[A, C]` with `fab >>> fbc`. If `F` is `Function1` then `>>>` becomes alias of `andThen`.
+- Taking two arrows `fab: F[A, B]` and `fbc: F[B, C]` and combining them into `F[(A, C) => (B, D)]` with `fab *** fbc`. The resulting arrow takes two inputs and processes them with two arrows, one for each input.
+- Taking two arrows `fab: F[A, B]` and `fac: F[A, C]` and combining them into `F[A => (B, C)]` with `fab &&& fbc`. The resulting arrow takes an input, duplicates it and processes each copy with a different arrow.
+- Taking an arrow `fab: F[A, B]` and turning it into `F[(C, A), (C, B)]` with `fab.second`. The resulting arrow takes two inputs, processes the second input and leaves the first input as it is.
+
+## Example
+
+As an example, let's create a fancy version of `Function1` that is capable of maintaining states. Let's call it `FancyFunction`:
+
+```tut:book:silent
+case class FancyFunction[A, B](run: A => (FancyFunction[A, B], B))
+```
+
+That is, given an `A`, it not only returns a `B`, but also returns a new `FancyFunction[A, B]`. This sounds similar as the `State` monad (which returns a result and a new `State` from an initial `State`), and indeed, `FancyFunction` can be used to perform stateful transformations.
+
+Here's an `Arrow` instance for `FancyFunction`:
+
+```tut:book:silent
+import cats.arrow.Arrow
+import cats.implicits._
+
+implicit val arrowInstance: Arrow[FancyFunction] = new Arrow[FancyFunction] {
+
+  override def lift[A, B](f: A => B): FancyFunction[A, B] = FancyFunction(lift(f) -> f(_))
+
+  override def first[A, B, C](fa: FancyFunction[A, B]): FancyFunction[(A, C), (B, C)] = FancyFunction {case (a, c) =>
+    val (fa2, b) = fa.run(a)
+    (first(fa2), (b, c))
+  }
+
+  override def id[A]: FancyFunction[A, A] = FancyFunction(id -> _)
+
+  override def compose[A, B, C](f: FancyFunction[B, C], g: FancyFunction[A, B]): FancyFunction[A, C] = FancyFunction {a =>
+    val (gg, b) = g.run(a)
+    val (ff, c) = f.run(b)
+    (compose(ff, gg), c)
+  }
+}
+
+```
+
+Then we can start to do interesting things with it. For example, we can create a method `accum` that returns a `FancyFunction`, which accumulates values fed to it using the accumulation function `f` and the starting value `b`:
+
+```tut:book:silent
+def accum[A, B](b: B)(f: (A, B) => B): FancyFunction[A, B] = FancyFunction {a =>
+  val b2 = f(a, b)
+  (accum(b2)(f), b2)
+}
+```
+
+We then define a method `runList` that takes a `List[A]`, feeds the `A` values one by one into the `FancyFunction` and collects the result into a `List[B]`:
+
+```tut:book:silent
+def runList[A, B](ff: FancyFunction[A, B], as: List[A]): List[B] = as match {
+  case h :: t =>
+    val (ff2, b) = ff.run(h)
+    b :: runList(ff2, t)
+  case _ => List()
+}
+```
+
+Now we can try it out:
+
+```tut:book:silent
+runList(accum[Int, Int](0)(_ + _), List(6, 5, 4, 3, 2, 1))
+// res1: List[Int] = List(6, 11, 15, 18, 20, 21)
```

Comment:
Can we not make this non-silent? Then we could remove the comment, or am I missing something?

Reply:
You are right, I'm not super familiar with tut.